File: node/algorithm/sum.js
// Count 1+..+n
// T(n) = O(n)
function sum(n) {
    let sum = 0
    let i = 1
    for (; i <= n; i++) {
        sum += i
    }
    return sum
}

// Count 
// T(n) = O(n^2)
function cal(n) {
    let sum = 0 // 1
    let i = 1   // 1 
    for (; i <= n; ++i) { // n
        let j = 1         // n
        for (; j <= n; ++j) { // n^2
            sum += i * j      // n^2
        }
    }
    return sum
}




// Main
// console.log(sum(5))
console.log(cal(2))

File: algorithm/time_complexity/sum.c
#include<stdio.h>

// T(n) = O(n)
int sum(int n){
    int sum = 0;
    for (int i=1;i<=n;i++){
        sum+=i;
    }    
    return sum;
}

// T(n) = O(n^2)
int cal(int n){
    int sum = 0;
    for (int i=1;i<=n;i++){
        for (int j=1;j<=n;j++){
            sum += i * j;
        }
    }
    return sum;
}

void main(){
    printf("%d\n",sum(5));
    printf("%d\n",cal(4));
}